File: 2215. Find the Difference of Two Arrays/find-the-difference-of-two-arrays.java
// class Solution {
//     public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
//         int[] diff_elem;
//         int[] diff_elem1;

//         Set<Integer> set = new HashSet<>();
//         Set<Integer> set1 = new HashSet<>();
        
//         for(int nums : nums1){
//             if(set1.find)
//         }
//     }
// }


class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        List<Integer> diff_elem = new ArrayList<>();
        List<Integer> diff_elem1 = new ArrayList<>();

        Set<Integer> set1 = new HashSet<>();
        Set<Integer> set2 = new HashSet<>();
        
        for (int num : nums1) {
            set1.add(num);
        }
        
        for (int num : nums2) {
            set2.add(num);
        }
        
        for (int num : nums1) {
            if (!set2.contains(num) && !diff_elem.contains(num)) {
                diff_elem.add(num);
            }
        }
        
        for (int num : nums2) {
            if (!set1.contains(num) && !diff_elem1.contains(num)) {
                diff_elem1.add(num);
            }
        }
        
        List<List<Integer>> result = new ArrayList<>();
        result.add(diff_elem);
        result.add(diff_elem1);
        
        return result;
    }
}

File: 2215. Find the Difference of Two Arrays/find-the-difference-of-two-arrays.js
var findDifference = function (nums1, nums2) {
    // Initialize arrays to store the differences
    const diff_elem = []; // Differences found in nums1 but not in nums2
    const diff_elem1 = []; // Differences found in nums2 but not in nums1

    // Initialize sets to keep track of numbers seen in each array
    const seennum = new Set(); // Set to store numbers seen in nums1
    const seennum1 = new Set(); // Set to store numbers seen in nums2

    // Iterate through nums1 to find differences
    nums1.forEach(num => {
        // Check if the current number is not present in nums2 and not already seen in nums1
        if (!nums2.includes(num) && !seennum.has(num)) {
            // Add the number to the differences array
            diff_elem.push(num);
            // Add the number to the set of seen numbers in nums1
            seennum.add(num);
        }
    });

    // Iterate through nums2 to find differences
    nums2.forEach(num => {
        // Check if the current number is not present in nums1 and not already seen in nums2
        if (!nums1.includes(num) && !seennum1.has(num)) {
            // Add the number to the differences array
            diff_elem1.push(num);
            // Add the number to the set of seen numbers in nums2
            seennum1.add(num);
        }
    });

    // Return the arrays containing the differences
    return [diff_elem, diff_elem1];
};
